sycl/source/function_pointer.cpp

```diff
@@ -16,13 +16,28 @@ namespace detail {
 ext::oneapi::device_func_ptr_holder_t
 getDeviceFunctionPointerImpl(device &D, program &P, const char *FuncName) {
   ext::oneapi::device_func_ptr_holder_t FPtr = 0;
-  // FIXME: return value must be checked here, but since we cannot yet check
-  // if corresponding extension is supported, let's silently ignore it here.
+
   const detail::plugin &Plugin = detail::getSyclObjImpl(P)->getPlugin();
-  Plugin.call<cl::sycl::detail::PiApiKind::piextGetDeviceFunctionPointer>(
+  pi_result Result = Plugin.call_nocheck<
+      cl::sycl::detail::PiApiKind::piextGetDeviceFunctionPointer>(
       detail::pi::cast<pi_device>(detail::getSyclObjImpl(D)->getHandleRef()),
       detail::pi::cast<pi_program>(detail::getSyclObjImpl(P)->getHandleRef()),
       FuncName, &FPtr);
+  // If extension is not supported, a fallback method is used which returns
+  // if the function exists or not. If the return address is not valid throw
+  // error.
+  if (Result != PI_SUCCESS) {
+    if (Result == PI_FUNCTION_ADDRESS_IS_NOT_AVAILABLE)
+      throw runtime_error("Function exists however, address is not available",
```

Comment:
Very minor but the use of "however", though correct, looks a bit out of place.

```suggestion
      throw runtime_error("Function exists but address is not available",
```